2018-02-11  Andrew Musk  <musk@lnxsrv04.seas.ucla.edu>

	* lib/timespec.h (timespec_cmp):
	Use comparisons instead of subtraction.
	This will avoid possible integer overflow.

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2018-02-11 16:11:54.094221000 -0700
@@ -45,8 +45,9 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if( a.tv_sec > b.tv_sec ) { return 1; }
+  else if( a.tv_sec < b.tv_sec ) { return -1; }
+  else { return 0; }
 }
 
 # if ! HAVE_DECL_NANOSLEEP